Makefile

```diff
@@ -4,12 +4,16 @@ dynamic: lib
 	$(AR) -x libmpfun90.a
 	$(CXX) -shared *.o -o libmpfun90.so
 
-	
+
 lib: libmpfun90.a
 
 
-FC = gfortran
-FFLAGS = -fno-automatic -O2 -funroll-all-loops -std=legacy -fPIC
+FC := $(shell echo $$FC)
+ifndef FC
+	FC := gfortran
+endif
+
+FFLAGS := $(shell echo $$FFLAGS) -fno-automatic -O2 -funroll-all-loops -std=legacy -fPIC
```

Comment:
That won't allow you to go to `-O3` if you like, so I'd rather put it behind.
Again with
```
FFLAGS = -fno-automatic -O2 -funroll-all-loops -std=legacy -fPIC $(FFLAGS)
```

You could call it with `make FFLAGS=${FFLAGS}`. To my understanding that is the default way Makefiles behave/are designed.

Reply:
For reference I found https://www.gnu.org/software/make/manual/html_node/Complex-Makefile.html where also `CC` and `CFLAGS` are directly set, but since we set some defaults lets prepend our default.

Reply:
Is there a reason to not just do

```make
FFLAGS ?= -fno-automatic -O2 -funroll-all-loops -std=legacy -fPIC
```

instead? (I'm asking here as I don't deeply know `Make` syntax beyond the basics.)

Reply:
My though process was to aim for as little modifactions necessary on the gentoo side, which automatically sets FFLAGS. By default e.g. `-std=legacy` is not set and then it would have to be set/append again on the ebuild/conda side.

Reply:
> By default e.g. `-std=legacy` is not set and then it would have to be set/appended again on the ebuild/conda side. So the question is are the FFLAGS on our side mandatory for the program to work (can't be overwritten) or optional/defaults (can be overwritten).

Good question RE: the `FFLAGS` which I don't know the answer to (which is part of the reason I opened Issue #2).

Do you know that for Gentoo all of 

https://github.com/APN-Pucky/mpfun90/blob/d00bfe06b8f69329464906817f7f8b795bed5388/Makefile#L12

are explicitly required? In the example below I'm using

```console
# echo $FC
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran
# echo $FFLAGS
-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include
```

and everything builds fine, as far as I can tell.

```console
$ docker run --rm -ti python:3.12 bash
root@7178225de27f:/# curl -fsSL https://pixi.sh/install.sh | bash && . /root/.bashrc
root@7178225de27f:/# mkdir -p example && cd $_
root@7178225de27f:/example# pixi init
✔ Created /example/pixi.toml
root@7178225de27f:/example# pixi add fortran-compiler make git
✔ Added fortran-compiler >=1.8.0,<2
✔ Added make >=4.4.1,<5
✔ Added git >=2.47.1,<3
root@7178225de27f:/example# pixi shell
(example) root@7178225de27f:/example# git clone https://github.com/matthewfeickert/mpfun90.git && cd mpfun90
(example) root@7178225de27f:/example/mpfun90# git checkout origin/feat/use-shell-information-for-defaults -b feat/use-shell-information-for-defaults
(example) root@7178225de27f:/example/mpfun90# echo $FC
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran
(example) root@7178225de27f:/example/mpfun90# echo $FFLAGS
-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include
(example) root@7178225de27f:/example/mpfun90# make dynamic
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpfun90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpfun90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar: creating libmpfun90.a
a - mpfun90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmod90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmod90.o
a - mpmod90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmodm90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmodm90.o
a - mpmodm90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmodx90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmodx90.o
a - mpmodx90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -x libmpfun90.a
g++ -shared *.o -o libmpfun90.so
(example) root@7178225de27f:/example/mpfun90# ls libmpfun90.*
libmpfun90.a  libmpfun90.so
(example) root@7178225de27f:/example/mpfun90#
```

We can also set your flags, and things build as well

```console
(example) root@7178225de27f:/example/mpfun90# rm -f *.a *.so *.o *.mod
(example) root@7178225de27f:/example/mpfun90# make dynamic FFLAGS="-fno-automatic -funroll-all-loops -std=legacy $FFLAGS"
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -fno-automatic -funroll-all-loops -std=legacy -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpfun90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpfun90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar: creating libmpfun90.a
a - mpfun90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -fno-automatic -funroll-all-loops -std=legacy -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmod90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmod90.o
a - mpmod90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -fno-automatic -funroll-all-loops -std=legacy -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmodm90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmodm90.o
a - mpmodm90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-gfortran -fno-automatic -funroll-all-loops -std=legacy -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /example/.pixi/envs/default/include -I/example/.pixi/envs/default/include -c mpmodx90.f90
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -rv libmpfun90.a mpmodx90.o
a - mpmodx90.o
/example/.pixi/envs/default/bin/x86_64-conda-linux-gnu-ar -x libmpfun90.a
g++ -shared *.o -o libmpfun90.so
(example) root@7178225de27f:/example/mpfun90# ls libmpfun90.*
libmpfun90.a  libmpfun90.so
(example) root@7178225de27f:/example/mpfun90#
```

Reply:
With `?=` instead of `=` I have to append `-fPIC`to `FFLAGS`. (`-fPIC` is needed for creating a `.so`, but not `.a`)

Reply:
On my Gentoo sytem I just have `FFLAGS=-O2 -pipe -frecord-gcc-switches` thus I need to add `-fPIC` (technically on Gentoo every user can/will customize their flags, but sth. light like mine is comon).

Reply:
I'd be in favor of dropping all but `-fPIC` so that we are consistent with mpfun2020 (very simple) compile script.

Reply:
> My thought process was to aim for as little modifications necessary on the gentoo side

Does Gentoo build from hashes? Or would any modification to this repository immediately get picked up and break builds?

It would be ideal in my mind if we could harcode the minimum set of required flags in the Makefile and have the rest be passed by the user to make things explicit about the build and not have people guess at flag choice. Though for https://github.com/conda-forge/staged-recipes/pull/28415 I'll just start doing

```console
make dynamic FC="$FC" FFLAGS="$FFLAGS"
```

Reply:
Gentoo uses the tagged release (I set the date YYYY.MM.DD as release version). It can be modified freely.

Reply:
I modified the repo directly now only one QA complaint remains:

```
 * QA Notice: Missing soname symlink(s):
 * 
 * 	usr/lib64/mpfun90.so.0 -> libmpfun90.so
 *
 ```
 
 should be easy enough to fix (Gentoo ideally want shared libraries to have a soname and since we are already modifying the Makefile here).